pegasus/router.rb

```diff
@@ -292,6 +292,8 @@ def document(path)
         cache :document
       end
 
+      pass if request.post? && !@header['post']
```

Comment:
`pass` would return a `404` error as long as there are no other routes further down in the route-handling chain for this Sinatra application (which there shouldn't be, since a template file was already found for this path at this point). I've changed the response code to 405 (`Method Not Allowed`) to be a bit more explicit. The code path is slightly different since `error` aborts the request immediately rather than passing it to the next route, but the two should have identical behavior in this case.